lib/auth/server.ts

```diff
@@ -93,18 +81,69 @@ export const awaitForToken = (options: {
 				if (!loggedIn) {
 					throw new ExpectedError('Invalid token');
 				}
+				this.emit('token', token);
 				response.status(200).render('success');
-				resolve(token);
 			} catch (error) {
+				this.emit('error', error);
 				response.status(401).render('error');
-				reject(new Error(error.message));
 			}
 		});
 
-		app.use((_request, response) => {
-			server.close(); // stop listening for new connections
+		this.expressApp.use((_request, response) => {
+			this.server.close(); // stop listening for new connections
+			this.emit('error', new Error('Unknown path or verb'));
 			response.status(404).send('Not found');
-			reject(new Error('Unknown path or verb'));
 		});
-	});
-};
+
+		return this.getAddress();
+	}
+
+	public getAddress(): { host: string; port: number; urlPath: string } {
+		const info = this.server.address() as import('net').AddressInfo;
+		return {
+			host: info.address,
+			port: info.port,
+			urlPath: this.loginPath,
+		};
+	}
+
+	/**
+	 * Shut the server down.
+	 * Call this method to avoid the process hanging in some situations.
+	 */
+	public shutdown() {
+		// By design (more like a bug, but they won't admit it), a Node.js `http.server`
+		// instance prevents the process from exiting for up to 2 minutes (by default) if a
+		// client keeps a HTTP connection open, and regardless of whether `server.close()`
+		// was called: the `server.close(callback)` callback takes just as long to be called.
+		// Setting `server.timeout` to some value like 3 seconds works, but then the CLI
+		// process hangs for "only" 3 seconds (not good enough). Reducing the timeout to 1
+		// second may cause authentication failure if the laptop or CI server are slow for
+		// any reason. The only reliable way around it seems to be to explicitly unref the
+		// sockets, so the event loop stops waiting for it. See:
+		// https://github.com/nodejs/node/issues/2642
+		// https://github.com/nodejs/node-v0.x-archive/issues/9066
+		//
+		this.serverSockets.forEach((s) => s.unref());
+		this.serverSockets.splice(0);
```

Comment:
You can replace all the code for this with https://www.npmjs.com/package/server-destroy

Reply:
The `server-destroy` package calls `socket.destroy()` on each socket, whereas this code calls `socket.unref()`. I am almost sure that I tried calling `socket.destroy()` and found that it was not useful / sufficient to fix issue #1404 (months ago), whereas calling `unref()` did the trick. You're welcome to try it. :-)
Also, _"all the code for this"_ is about 3 lines of code (excluding comments), and adding a dependency to `package.json` and importing and calling it would also be about 3 lines of code...